_episodes/01-connecting.md

```diff
@@ -129,6 +128,56 @@ For those logging in with PuTTY it would likely be best to cover the terminal
 basics already mentioned above before moving on to navigating the remote
 system.
 
+## Creating an SSH key
+
+SSH keys are an alternative method for authentication to obtain access to 
+remote computing systems. They can also be used for authentication when
+transferring files or for accessing version control systems. In this section
+you will create a pair of SSH keys, a private key which you keep on your
+own computer and a public key which is placed on the remote HPC system
+that you will log in to.
+
+### Linux, Mac and Windows Subsystem for Linux
+
+Once you have opened a terminal generate a public private SSH key pair by
+using 
+
+```
+ssh-keygen -o -a 100 -t rsa -b 4096 -f ~/.ssh/key_for_remote_computer
```

Comment:
Recommend explaining some non-default (but recommended) flags. RSA is the default and certainly widely supported, but a 4096-bit key provides no practical advantage. Upgrading to a more modern algorithm is better practice. The recommended function call [[1](https://security.stackexchange.com/a/144044), [2](https://docs.borgbase.com/faq/ssh/#step-2--create-keypair), [3](https://stribika.github.io/2015/01/04/secure-secure-shell.html)] uses the Edwards curve Digital Signature Algorithm.

```bash
ssh-keygen -t ed25519 -a 100
```

~~~markdown
Check for existing SSH keys and filenames,
then generate a new public-private keypair.

```
$ ls ~/.ssh/
$ ssh-keygen -t ed25519 -a 100 -f ~/.ssh/id_ed25519
```
{: .language-bash}

* `-o` (no default): use the OpenSSH key format,
  rather than PEM.
* `-a` (default is 16): number of rounds of passphrase derivation;
  increase to slow down brute force attacks.
* `-t` (default is [rsa](https://en.wikipedia.org/wiki/RSA_(cryptosystem))):
  specify the "type" or cryptographic algorithm. 
  [ed25519](https://en.wikipedia.org/wiki/EdDSA)
 is faster and shorter for comparable strength.
* `-f` (default is /home/user/.ssh/id_algo): filename to store your keys.
  If you already have SSH keys, make sure you specify a different name:
  `ssh-keygen` will overwrite the default key if you don't specify!

If EcDSA is not available, use the older (but strong and trusted)
[RSA](https://en.wikipedia.org/wiki/RSA_(cryptosystem)) cryptography:

```
$ ls ~/.ssh/
$ ssh-keygen -o -a 100 -t rsa -b 4096 -f ~/.ssh/id_rsa
```
{: .language-bash}
 
The flag `-b` sets the number of bits in the key.
The default is 2048. EdDSA uses a fixed key length,
so this flag would have no effect.

> *N.B.:* For a deeper dive on SSH security and some of the
> flags set here, an excellent resource is [Secure Secure Shell](
> https://stribika.github.io/2015/01/04/secure-secure-shell.html).
~~~

Reply:
Ok. Thanks.

Reply:
Sorry, re-reading, this is unclear: I meant the larger block as a suggested replacement for the `ssh-keygen` command and explanation, it is not an extended quote from the reference materials. If you like it, please use it!